### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by MrChausson** [Starystars67/BeamNG-MP#383]
**Is your feature request related to a problem? Please describe.**
Hello guys I have setup a BeamNG server on a VPS that only support IPv6 for port forwarding. But the BeamMP client refuses to connect when I enter a IPv6 address in the quickconnect/direct connect tab.

**Describe the solution you'd like**
As IPv6 becomes more and more common (and it is so cool because opening a port on a IPv6 is basically free, in the other hand having an dedicated IPv4 is quite expensive). I think it would be wise to plan support for IPv6 connectivity very soon.

It is written in the wiki that IPv6 support is planned but there is no date when this will be available ?
Do you know this scale of this feature ? Is it difficult to implement in the existing codebase ?

**Describe alternatives you've considered**
An alternative solution is using tailscale, but it is only fine for private servers with friends as you have to invite every one of them to be able to connect to your server.

Thanks

**Comment by lionkor** [comment]
IPv6 is not supported at the moment.

> Is it difficult to implement in the existing codebase ?

No, it's a 2-4 line (or so) change in terms of Launcher and Server. The backend and mod would likely need changes, too, but those can't be complicated either.

IPv6 support is very easy to do and should be done by default in any codebase IMO.

so TODO for this issue would be

- [ ] Server IPv6 support (probably already supports it, but consult boost docs)
- [ ] Launcher IPv6 support (via protocol/socket type INET6)
- [ ] Backend changes if needed
- [ ] Mod changes if needed

**Comment by delta-whiplash** [comment]
This could be insane to implement this feature am searching for this for years 😅

**Comment by Starystars67** [comment]
> IPv6 is not supported at the moment.
> 
> > Is it difficult to implement in the existing codebase ?
> 
> No, it's a 2-4 line (or so) change in terms of Launcher and Server. The backend and mod would likely need changes, too, but those can't be complicated either.
> 
> IPv6 support is very easy to do and should be done by default in any codebase IMO.
> 
> so TODO for this issue would be
> 
> * [ ]  Server IPv6 support (probably already supports it, but consult boost docs)
> * [ ]  Launcher IPv6 support (via protocol/socket type INET6)
> * [ ]  Backend changes if needed
> * [ ]  Mod changes if needed

Agreed backend changes would be approx 4 lines of changes. I can look into adding this later today.

It may however also require some adjustment to our load balancers too @Mack29446 thoughts on needs to forward the Cloudflare client IPv6?

@lionkor do you think it would be possible to have the server tell the backend if it prefers IPv6 in its heartbeat?

**Comment by lionkor** [comment]
I think if IPv4 is available, always use that, if its not, use IPv6, sounds solid to me

**Comment by lionkor** [comment]
Though it would probably be on @Anonymous-275 to implement, so would love to hear his thoughts

**Comment by Starystars67** [comment]
So, so far the infrastructure does support IPv6 but it will prefer IPv4 if available.

Given this I think it is just Server/Launcher/Lua changes for this now?

**Comment by MrChausson** [comment]
Hello ! Do you have any news for this ? Do you know who would be able to make the mod modification ?
